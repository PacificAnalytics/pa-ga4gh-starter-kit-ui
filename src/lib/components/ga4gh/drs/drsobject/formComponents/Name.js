import React from 'react';
import { SimpleTextField } from '../../../../common/form';

const Name = props => {
    return (
        <SimpleTextField
            id="name"
            label="Name"
            name="name"
            value={props.name}
            readOnly={props.readOnly}
            changeFunction={props.setName}
            helperText='Short, descriptive name for this DRS Object.'
        />
    )
}

export default Name;