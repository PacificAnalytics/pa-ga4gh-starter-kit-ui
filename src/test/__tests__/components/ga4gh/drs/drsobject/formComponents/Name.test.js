import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import {mockUpdateScalar} from '../../../../../../resources/MockFunctions';
import Name from '../../../../../../../lib/components/ga4gh/drs/drsobject/formComponents/Name';

afterEach(() => {
    mockUpdateScalar.mockClear();
})

test('SHOW <Name /> should handle populated field', () => {
    let {container} = render(
        <Name readOnly={true} name='test name' setName={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Name')).toBeInTheDocument();
    let nameField = screen.getByRole('textbox');
    expect(nameField).toHaveValue('test name');

    // name field is read only and cannot be edited
    expect(nameField).toHaveAttribute('readonly');
    userEvent.type(nameField, 'new value');
    expect(mockUpdateScalar.mock.calls.length).toBe(0);
});

test('NEW and EDIT <Name /> should handle empty field', () => {
    let {container} = render(
        <Name readOnly={false} name='' setName={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Name')).toBeInTheDocument();
    let nameField = screen.getByRole('textbox');
    expect(nameField).toHaveValue('');
    userEvent.type(nameField, 'new value');
    expect(mockUpdateScalar).toHaveBeenCalled();
});

test('NEW and EDIT <Name /> should handle populated field', () => {
    let {container} = render(
        <Name readOnly={false} name='test name' setName={mockUpdateScalar} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('Name')).toBeInTheDocument();
    let nameField = screen.getByRole('textbox');
    expect(nameField).toHaveValue('test name');
    userEvent.type(nameField, 'new value');
    expect(mockUpdateScalar).toHaveBeenCalled();
});