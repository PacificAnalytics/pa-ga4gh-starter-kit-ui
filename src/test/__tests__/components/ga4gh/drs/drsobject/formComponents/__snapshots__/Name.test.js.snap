// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <Name /> should handle empty field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      for="name"
      id="name-label"
    >
      Name
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="name-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="name"
        name="name"
        type="text"
        value=""
      />
    </div>
    <p
      class="MuiFormHelperText-root"
      id="name-helper-text"
    >
      Short, descriptive name for this DRS Object.
    </p>
  </div>
</div>
`;

exports[`NEW and EDIT <Name /> should handle populated field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="name"
      id="name-label"
    >
      Name
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="name-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="name"
        name="name"
        type="text"
        value="test name"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="name-helper-text"
    >
      Short, descriptive name for this DRS Object.
    </p>
  </div>
</div>
`;

exports[`SHOW <Name /> should handle populated field 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="name"
      id="name-label"
    >
      Name
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-describedby="name-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="name"
        name="name"
        readonly=""
        type="text"
        value="test name"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-filled"
      id="name-helper-text"
    >
      Short, descriptive name for this DRS Object.
    </p>
  </div>
</div>
`;
